sklearn/base.py

```diff
@@ -465,7 +467,12 @@ def _repr_mimebundle_(self, **kwargs):
 class ClassifierMixin:
     """Mixin class for all classifiers in scikit-learn."""
 
-    _estimator_type = "classifier"
+    # mypy error: Decorated property not supported
+    @deprecated("_estimator_type is deprecated in "  # type: ignore
+                "0.24 and will be removed in 0.26")
+    @property
+    def _estimator_type(self):
+        return "classifier"
```

Comment:
I am +0.1 regarding deprecating a private property.

Reply:
I do not like to deprecate a private property neither. However, it may be weird for downstream dependencies (e.g. `XGBoost` or `LightGBM`) to remove this private property without deprecating if they are using it.

WDYT?

Reply:
After investigating, [`XGBoost` is using `ClassifierMixin`](https://github.com/dmlc/xgboost/blob/master/python-package/xgboost/compat.py#L127). Then, I would be happy to remove the deprecation of `_estimator_type`.

Thanks @thomasjpfan and @NicolasHug for the suggestion!